src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,49 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    if (!Animal.alive) {
+      Animal.alive = [];
+    }
+
```

Comment:
To avoid splicing the alive array when the animal is dead, let's introduce a static method `removeFromAlive` or something like that, this should incapsulate this logic within the Animal class.